fedimint-server/src/config.rs

```diff
@@ -168,14 +168,26 @@ impl ServerConfigConsensus {
         })
     }
 
-    pub fn to_client_config(&self, module_config_gens: &ModuleConfigGens) -> ClientConfig {
+    pub fn to_client_config(&self, module_config_gens: &ModuleInitRegistry) -> ClientConfig {
         self.to_client_config_try(module_config_gens)
             .expect("configuration mismatch")
     }
 }
+use impl_tools::autoimpl;
 
-pub type ModuleConfigGens =
-    BTreeMap<&'static str, Arc<dyn FederationModuleConfigGen + Send + Sync>>;
+// TODO: turn into newtype
+pub type DynModuleInit = Arc<dyn ModuleInit + Send + Sync>;
+
+// TODO: turn all registeries into `Registry<T>(BTreeMap<&'static str, T))` + type alias
+#[autoimpl(Deref using self.0)]
+#[derive(Clone)]
+pub struct ModuleInitRegistry(BTreeMap<&'static str, DynModuleInit>);
+
+impl<const N: usize> From<[(&'static str, DynModuleInit); N]> for ModuleInitRegistry {
+    fn from(value: [(&'static str, DynModuleInit); N]) -> Self {
+        Self(BTreeMap::from(value))
+    }
+}
```

Comment:
Could we make this `From<[DynModuleInit]>` by giving ´ModuleInit´ a `fn module_name(&self) -> &static str`?

Reply:
I should have made it a a `IntoIterator` to being with.

Reply:
#1231 This PR is already getting too big. :D